File: python/path3.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def pathSum(self, root: Optional[TreeNode], targetSum: int) -> int:
        self.result=0
        self.ts=targetSum

        def dfs(node,path):
            if node:
                path.append(node.val)
                for i in range(len(path)):
                    subpath=path[i:]
                    if checksum(subpath):
                        self.result+=1
                
                dfs(node.left,path)
                dfs(node.right,path)
                path.pop()
        def checksum(path):
            return sum(path)==self.ts
                    
        dfs(root,[])
        return self.result
            

File: javascript/path3.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} targetSum
 * @return {number}
 */
var pathSum = function(root, targetSum) {
    let result=0;
    function dfs(node,path){

        if (node!=null){
            path.push(node.val);
            for (let i=0;i<path.length;i++){
                
                subpath=path.slice(i);
                const s=subpath.reduce((agg,curr)=> agg+curr);
                if (s===targetSum){
                    result++;
                }
            }

            dfs(node.left,path);
            dfs(node.right,path);
            path.pop();
        }
    }
    dfs(root,[]);
    return result
    
};